lib/index.ts

```diff
@@ -0,0 +1,90 @@
+import * as _ from 'lodash';
+import { parse, Node } from './indent-parser';
+
+export { parseLockFile };
+
+export interface PkgNode {
+  name: string;
+  version: string;
+  restriction: string;
+  // ignoring for now - addressing later
+  // depType?: DepType;
+  // targetFrameworks?: string[];
+}
+
+export interface PkgTree {
+  type: nuget;
+  name: string;
+  dependencies: {
+    [dep: string]: PkgNode;
+  };
+  hasDevDependencies?: boolean;
+}
+
+const REPO_TYPES = ['NUGET'];
+
+// as with other dotnet-deps-parser we assume this parser will get the manifestFileContents as a string
+function parseLockFile(manifestFileContents: string) {
+  const depTree: PkgTree = {
+    dependencies: {},
+    hasDevDependencies: false,
+    name: '',
+  };
+  // assume first content is always Main GROUP
+  // parse from first line to next GROUP we find
+  // perhaps we grab more intelligently around NUGET alone to help performance?
+  const groups: string[] = manifestFileContents.split(/[Gg][Rr][Oo][Uu][Pp] +/).
+  // @ts-ignore
+  console.log(groups);
+  for (const group of groups) {
+    // first word in every group but the first one will be the name of the group -
+    // make sure to check this when we support devDependencies separately from main deps
+
+    // NUGET
+    //   remote: https://nuget.org/api/v2
+    //     FSharp.Core (4.0.0.1)
+    //     Newtonsoft.Json (7.0.1)
+    //       UnionArgParser (>=0.6.3)
+    //     UnionArgParser (0.6.3)
+    // GITHUB
+    //   remote: forki/FsUnit
+
+    for (const repoType of REPO_TYPES) {
+      // from line with no indent to next line with no indent
+      const repo = group
+        .substring(group.indexOf(repoType))
+        .split(/(?:^\w+\n)|(?:\n\w+)/)[1];
+      const parsedRepo = parse(repo);
+
+    }
+
+  }
+
+  for (const group of groups) {
+    const parsedGroup = parse(group);
+    const currentDepGroup = parsedGroup.children;
+
+    // absolutely not ideal another traversal to get NUGET
+    // logic works for MVP only
+    // TODO: make more performant
+    _.forEach(currentDepGroup['NUGET'], (repo) => {
+      _.forEach(repo, (dep, key: string) => {
+        const parsed = parseLockFileRow(key);
+        depTree.dependencies[parsed.name] = parsed;
+      });
+    });
+  }
+
+  return depTree;
+}
+
+function parseLockFileRow(row: string): PkgNode {
+  const rowParts = row.trim().split(' - restriction: ');
```

Comment:
We should split by ` - `  as everything that follows the `-` can be an array of options e.g.:
```
NUGET
  remote: https://api.nuget.org/v3/index.json
    NUnit (3.11.0) - version_in_path: true, restriction: == net451
```